brody_lab_to_nwb/interfaces/protocol_info/protocol_info_utils.py

```diff
@@ -0,0 +1,192 @@
+
+    # get n_trial length items into df
+    beh_df['delay'] = beh_info['delay']
+    beh_df['pair_hist'] = beh_info['pair_history']
+    beh_df['correct_side'] = beh_info['correct_side']
+    beh_df['prev_side'] = prev_side_adj
+    beh_df['aud1_sigma'] = beh_info['aud1_sigma']
+    beh_df['aud2_sigma'] = beh_info['aud2_sigma']
+
+    # extract parsed events/state machine info for each trial
+    parsed_events_dict = beh_info["parsed_events"]
+
+    # initilize space
+    c_poke = np.zeros((len(parsed_events_dict)))
+    hit_state = np.zeros((len(parsed_events_dict)))
+    aud1_on = np.zeros((len(parsed_events_dict)))
+    aud1_off = np.zeros((len(parsed_events_dict)))
+    aud2_on = np.zeros((len(parsed_events_dict)))
+    aud2_off = np.zeros((len(parsed_events_dict)))
+    end_state = np.zeros((len(parsed_events_dict)))
+
+    # iterate over items from state matrix
+    for trial in range(len(parsed_events_dict)):
+
+        # every trial has a center poke & end_state
+        c_poke[trial] = parsed_events_dict[trial]['states']['cp'][0]
+        end_state[trial] = parsed_events_dict[trial]['states']['check_next_trial_ready'][1]
+
+        # not all trials will have sound/hit time/etc, pull out info for non-violated
+        if beh_df['hit_hist'][trial] == 'viol':
+            hit_state[trial] = float("NaN")
+            aud1_on[trial] = float("NaN")
+            aud1_off[trial] = float("NaN")
+            aud2_on[trial] = float("NaN")
+            aud2_off[trial] = float("NaN")
+
+        elif beh_df['hit_hist'][trial] == 'hit':
+            hit_state[trial] = parsed_events_dict[trial]['states']['hit_state'][0]
+            aud1_on[trial] = parsed_events_dict[trial]['waves']['stimAUD1'][0]
+            aud1_off[trial] = parsed_events_dict[trial]['waves']['stimAUD1'][1]
+            aud2_on[trial] = parsed_events_dict[trial]['waves']['stimAUD2'][0]
+            aud2_off[trial] = parsed_events_dict[trial]['waves']['stimAUD2'][1]
+
+        elif beh_df['hit_hist'][trial] == 'miss':
+            hit_state[trial] = parsed_events_dict[trial]['states']['second_hit_state'][0]
+            aud1_on[trial] = parsed_events_dict[trial]['waves']['stimAUD1'][0]
+            aud1_off[trial] = parsed_events_dict[trial]['waves']['stimAUD1'][1]
+            aud2_on[trial] = parsed_events_dict[trial]['waves']['stimAUD2'][0]
+            aud2_off[trial] = parsed_events_dict[trial]['waves']['stimAUD2'][1]
+
+        else:
+            raise Exception('hit_hist doesn''t appear to have correct structure (hit, miss, viol)')
+
+    beh_df['c_poke'] = c_poke
+    beh_df['end_state'] = end_state
+    beh_df['hit_state'] = hit_state
+    beh_df['aud1_on'] = aud1_on
+    beh_df['aud1_off'] = aud1_off
+    beh_df['aud2_on'] = aud2_on
+    beh_df['aud2_off'] = aud2_off
+    beh_df['end_state'] = end_state
+
+    find_loudness(beh_df)
+    find_first_sound(beh_df)
+
+    pd.options.mode.chained_assignment = 'warn'
```

Comment:
what does this do?

Reply:
Not sure, everything in these utils is from https://github.com/Brody-Lab/jbreda_PWM_mPFC_analysis/blob/main/io_utils.py so maybe @jess-breda can explain